src/main/java/io/fabric8/maven/docker/AbstractBuildSupportMojo.java

```diff
@@ -63,7 +68,27 @@ protected void buildImage(ServiceHub hub, ImageConfiguration imageConfig)
         autoPullBaseImage(hub, imageConfig);
 
         MojoParameters params = createMojoParameters();
-        hub.getBuildService().buildImage(imageConfig, params, checkForNocache(imageConfig));
+        hub.getBuildService().buildImage(imageConfig, params, checkForNocache(imageConfig), addBuildArgs());
+    }
+
+    private Map<String, String> addBuildArgs() {
+        Map<String, String> buildArgsFromProject = addBuildArgsFromProperties(project.getProperties());
+        Map<String, String> buildArgsFromSystem = addBuildArgsFromProperties(System.getProperties());
+        return ImmutableMap.<String, String>builder().putAll(buildArgsFromProject).putAll(buildArgsFromSystem).build();
+    }
+
+    private Map<String, String> addBuildArgsFromProperties(Properties properties) {
+        String argPrefix  = "docker.buildArg.";
+        Map<String, String> buildArgs = new HashMap<>();
+        for(Object keyObj : properties.keySet()){
+            String key = (String)keyObj;
+            if (key.startsWith(argPrefix)){
+                String argKey = key.replace(argPrefix, "");
```

Comment:
would use either `replaceFirst` or `substring(argPrefix.length()` for the unlikely case that someone use a strange name for its build arg :)